### PR TITLE
Market decentralized/agreement approve

### LINKED ISSUE #382: Approve Agreement
- [ ] Implement Provider POST `agreements/{agreement_id}/approve` endpoint
    - [ ] Change local Agreement state to approved
    - [ ] Send message to Requestor
    - [ ] Requestor marks Agreement as Approved

If Requestor waited on POST `agreements/{agreement_id}/wait` it will be handled in different task

### PR DESCRIPTION
**Pull request by tworec**
resolves #382

### CODE REVIEW

File: core/market/decentralized/src/db/model/agreement.rs [hunk 1]
```diff
@@ -111,6 +113,53 @@ impl Agreement {
...
+            approved_date: self
+                .approved_date
+                .map(|d| DateTime::<Utc>::from_utc(d, Utc)),
+            state: self.state.into(),
+            proposed_signature: self.proposed_signature,
+            approved_signature: self.approved_signature,
+            committed_signature: self.committed_signature,
+        })
+    }
+}
```
Comment (nieznanysprawiciel) [inline core/market/decentralized/src/db/model/agreement.rs:149]: We wrote the same with @jiivan in get gsb agreement :(

File: core/market/decentralized/src/db/model/agreement.rs [hunk 2]
```diff
@@ -111,6 +113,53 @@ impl Agreement {
...
+    pub fn into_client(self) -> Result<ClientAgreement, ErrorMessage> {
+        let demand_properties = serde_json::from_str(&self.demand_properties)
+            .map_err(|e| format!("Can't serialize Demand properties. Error: {}", e))?;
+        let offer_properties = serde_json::from_str(&self.offer_properties)
+            .map_err(|e| format!("Can't serialize Offer properties. Error: {}", e))?;
+
+        let demand = Demand {
+            properties: demand_properties,
+            constraints: self.demand_constraints,
+            demand_id: Some(self.demand_id.to_string()),
```
Comment (nieznanysprawiciel) [inline core/market/decentralized/src/db/model/agreement.rs:126]: Misleading: properties aren't related to demand, but to last proposals

File: core/market/decentralized/src/db/model/agreement.rs [hunk 3]
```diff
@@ -111,6 +113,53 @@ impl Agreement {
...
+            requestor_id: Some(self.requestor_id.to_string()),
+        };
+        let offer = Offer {
+            properties: offer_properties,
+            constraints: self.offer_constraints,
+            offer_id: Some(self.offer_id.to_string()),
```
Comment (nieznanysprawiciel) [inline core/market/decentralized/src/db/model/agreement.rs:132]: Misleading: properties aren't related to offer, but to last proposals

File: core/market/decentralized/src/db/model/events.rs [hunk 4]
```diff
@@ -48,15 +49,15 @@ pub struct MarketEvent {
...
+    pub artifact_id: ProposalId,
 }
 
 #[derive(Clone, Debug, Insertable)]
 #[table_name = "market_event"]
 pub struct NewMarketEvent {
     pub subscription_id: SubscriptionId,
     pub event_type: EventType,
-    pub artifact_id: String, // TODO: typed
+    pub artifact_id: ProposalId, // TODO: typed
```
Comment (nieznanysprawiciel) [inline core/market/decentralized/src/db/model/events.rs:60]: What with PropertyQuery?

File: core/market/decentralized/src/db/model/events.rs [hunk 5]
```diff
@@ -99,17 +100,25 @@ impl MarketEvent {
...
 
         Ok(prop.into_client()?)
     }
 
+    async fn into_client_agreement(self, db: DbExecutor) -> Result<ClientAgreement, EventError> {
+        let prop = db
+            .as_dao::<AgreementDao>()
+            .select(&self.artifact_id, Utc::now().naive_utc())
+            .await
+            .map_err(|error| EventError::FailedGetProposal(error))?
```
Comment (nieznanysprawiciel) [inline core/market/decentralized/src/db/model/events.rs:116]: FailedGetProposal not Agreement?

File: core/market/decentralized/src/db/model/agreement.rs (hunk 1 above)
Comment (tworec) [reply core/market/decentralized/src/db/model/agreement.rs:149]: :(

File: core/market/decentralized/src/db/model/events.rs (hunk 4 above)
Comment (tworec) [reply core/market/decentralized/src/db/model/events.rs:60]: it is not in scope of MVP. So let's not bother now

File: core/market/decentralized/src/db/model/agreement.rs (hunk 2 above)
Comment (tworec) [reply core/market/decentralized/src/db/model/agreement.rs:126]: but this comment relates more to `ya-client`

File: core/market/decentralized/src/db/model/events.rs (hunk 4 above)
Comment (nieznanysprawiciel) [reply core/market/decentralized/src/db/model/events.rs:60]: That's why I don't think you should change it

**Pull request merged**